inlong-sort/sort-connectors/elasticsearch-base/src/main/java/org/apache/inlong/sort/elasticsearch/table/MultipleElasticsearchSinkFunctionBase.java

```diff
@@ -0,0 +1,297 @@
+        this.createKey = Preconditions.checkNotNull(createKey);
+        this.createRouting = createRouting;
+        this.dirtySinkHelper = dirtySinkHelper;
+        this.tableSchemaFactory = tableSchemaFactory;
+        this.multipleFormat = multipleFormat;
+        this.indexPattern = indexPattern;
+        this.schemaUpdateExceptionPolicy = schemaUpdateExceptionPolicy;
+    }
+
+    @Override
+    public void open(RuntimeContext ctx, SinkMetricData sinkMetricData) {
+        indexGeneratorMap = new HashMap<>();
+        this.sinkMetricData = sinkMetricData;
+    }
+
+    private void sendMetrics(byte[] document) {
+        if (sinkMetricData != null) {
+            sinkMetricData.invoke(1, document.length);
+        }
+    }
+
+    @Override
+    public void initializeState(FunctionInitializationContext context) {
+    }
+
+    @Override
+    public void snapshotState(FunctionSnapshotContext context) {
+    }
+
+    @Override
+    public void process(RowData element, RuntimeContext ctx, RequestIndexer<Request> indexer) {
+        JsonNode rootNode = null;
+        // parse rootnode
+        try {
+            jsonDynamicSchemaFormat =
+                    (JsonDynamicSchemaFormat) DynamicSchemaFormatFactory.getFormat(multipleFormat);
+            rootNode = jsonDynamicSchemaFormat.deserialize(element.getBinary(0));
+            // Ignore ddl change for now
+            boolean isDDL = jsonDynamicSchemaFormat.extractDDLFlag(rootNode);
+            if (isDDL) {
+                LOGGER.error("ddl change unsupported");
+                return;
+            }
+        } catch (Exception e) {
+            LOGGER.error(String.format("deserialize error, raw data: %s", new String(element.getBinary(0))), e);
+        }
+
+        RowType rowType = jsonDynamicSchemaFormat.extractSchema(rootNode);
+        RowData data = jsonDynamicSchemaFormat.extractRowData(rootNode, rowType).get(0);
+        // generate the serialization schema
+        serializationSchema = new JsonRowDataSerializationSchema(
+                rowType, TimestampFormat.ISO_8601, MapNullKeyMode.LITERAL, "null", true);
+
+        final byte[] document;
+        try {
+            // for multiple sink, need to update runtimeconverter to correct rowtype
+            // for now create custom schema
+            document = serializationSchema.serialize(data);
+        } catch (Exception e) {
+            LOGGER.error(String.format("Serialize error, raw data: %s", data), e);
+            dirtySinkHelper.invoke(data, DirtyType.SERIALIZE_ERROR, e);
+            if (sinkMetricData != null) {
+                sinkMetricData.invokeDirty(1, data.toString().getBytes(StandardCharsets.UTF_8).length);
+            }
+            return;
+        }
+        final String key;
+        try {
+            // use uuid3 as key
+            JsonNode physicalData = jsonDynamicSchemaFormat.getPhysicalData(rootNode);
+            // will this line have performance issues?
```

Comment:
what does this comment stands for ?

Reply:
this line might introduce performance issues, since generating an uuid for every rowdata might spend too much time. 
If performance is too low, then we might just use physicaldata.hashcode()